### DISCUSSION TITLE
How Can Create Ant design Persian Date Picker (jalali)

### DISCUSSION THREAD

**Discussion by mahdiporkar** [ant-design/ant-design#51311] [General]
### What problem does this feature solve?

i saw in document we can set local fa but this cause title of Date Picker  change to persian but we need DatePicker change to jalali

### What does the proposed API look like?

.

**Comment by petercat-assistant** [comment]
To create a Persian Date Picker (Jalali) in Ant Design, you can follow these steps:

1. **Use a Jalali Date Library**: You need to use a library that supports Jalali dates, such as `moment-jalaali` or `dayjs` with a Jalali plugin.

2. **Configure Ant Design DatePicker**: Ant Design's DatePicker can be configured to use a custom date library. You need to integrate the Jalali date library with Ant Design's DatePicker.

3. **Set Locale**: While you can set the locale to 'fa' for Persian, you need to ensure that the date calculations and formatting are done using the Jalali calendar.

Here's a basic example using `moment-jalaali`:

```javascript
import React from 'react';
import { DatePicker, ConfigProvider } from 'antd';
import moment from 'moment-jalaali';
import 'moment/locale/fa';
import fa_IR from 'antd/lib/locale/fa_IR';

// Set moment to use Jalali
moment.loadPersian({ dialect: 'persian-modern', useGregorianParser: true });

const JalaliDatePicker = () => {
  return (
    <ConfigProvider locale={fa_IR}>
      <DatePicker
        picker="date"
        value={moment()}
        onChange={(date) => console.log(date ? date.format('jYYYY/jMM/jDD') : '')}
      />
    </ConfigProvider>
  );
};

export default JalaliDatePicker;
```

### Key Points:
- **Locale**: Set the locale to Persian using Ant Design's `ConfigProvider`.
- **Jalali Dates**: Use `moment-jalaali` to handle Jalali date formatting and parsing.
- **DatePicker**: Ensure the DatePicker is configured to use the Jalali date format.

This setup should help you create a DatePicker that uses the Jalali calendar while displaying in Persian. If you encounter any issues or need further customization, feel free to ask!

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by mahdiporkar** [comment]
sorry this code not worked

**Comment by petercat-assistant** [comment]
To create a Persian Date Picker (Jalali) using Ant Design, you can follow these steps:

1. **Install Jalali Date Library**: You need a library that supports Jalali dates, such as `moment-jalaali` or `dayjs` with Jalali plugin.

2. **Configure Ant Design DatePicker**: Use the `DatePicker` component from Ant Design and configure it to use the Jalali date library.

3. **Set Locale**: Ensure that the locale is set to Persian (`fa`) to display the date in the Persian format.

Here's a basic example using `moment-jalaali`:

```javascript
import React from 'react';
import { DatePicker, ConfigProvider } from 'antd';
import 'moment/locale/fa';
import moment from 'moment-jalaali';

// Set the locale globally
moment.loadPersian({ dialect: 'persian-modern', useGregorianParser: true });

const JalaliDatePicker = () => {
  return (
    <ConfigProvider locale={{ locale: 'fa' }}>
      <DatePicker
        defaultValue={moment()}
        onChange={(date, dateString) => console.log(date, dateString)}
      />
    </ConfigProvider>
  );
};

export default JalaliDatePicker;
```

### Steps to Implement:

1. **Install Dependencies**:
   ```bash
   npm install moment moment-jalaali antd
   ```

2. **Import and Configure**:
   - Import `moment-jalaali` and configure it to use the Jalali calendar.
   - Use `ConfigProvider` from Ant Design to set the locale to Persian.

3. **Use DatePicker**:
   - Use the `DatePicker` component and handle the date change as needed.

This setup will allow you to use the Ant Design DatePicker with Jalali dates. If you need further customization or encounter any issues, feel free to ask for more assistance.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Reply by mahdiporkar** [reply]
your guide not working

**Reply by mahdiporkar** [reply]
your code not working

**Comment by HrRafiei** [comment]
npm install antd-jalali dayjs jalali-dayjs

then: 
import React from 'react';
import { ConfigProvider } from 'antd';
import { DatePicker } from 'antd-jalali';
import fa_IR from 'antd/lib/locale/fa_IR';
import 'dayjs/locale/fa'; // برای لوکال فارسی
import jalaliPlugin from 'jalali-dayjs';

import dayjs from 'dayjs';
dayjs.extend(jalaliPlugin);
dayjs.locale('fa'); // تنظیم زبان به فارسی

export default function TestPage() {
  return (
    <div className="p-5">
      <ConfigProvider direction="rtl" locale={fa_IR}>
        <DatePicker style={{ width: 200 }} placeholder="تاریخ را انتخاب کنید" />
      </ConfigProvider>
    </div>
  );
}




But we will have a bug with month, Their nouns will be uncompleted
